File: example/javabuilds/4-nested-modules/baz/src/Baz.scala
package baz
import scalatags.Text.all._
import mainargs.{main, ParserForMethods, arg}

object Baz {
  @main
  def main(@arg(name = "bar-text") barText: String,
           @arg(name = "qux-text") quxText: String,
           @arg(name = "foo-text") fooText: String,
           @arg(name = "baz-text") bazText: String): Unit = {
    foo.Foo.main(barText, quxText, fooText)

    val value = p(bazText)
    println("Baz.value: " + value)
  }

  def main(args: Array[String]): Unit = ParserForMethods(this).runOrExit(args)
}

File: example/javabuilds/4-nested-modules/baz/src/baz/Baz.java
package baz;

import org.apache.commons.text.StringEscapeUtils;
import net.sourceforge.argparse4j.ArgumentParsers;
import net.sourceforge.argparse4j.inf.ArgumentParser;
import net.sourceforge.argparse4j.inf.Namespace;
import foo.Foo;

public class Baz {

  public static void main(String barText, String quxText, String fooText, String bazText) {
    Foo.main(barText, quxText, fooText);

    String value = "<p>" + StringEscapeUtils.escapeHtml4(bazText) + "</p>";
    System.out.println("Baz.value: " + value);
  }

  public static void main(String[] args) throws Exception {
    ArgumentParser parser = ArgumentParsers.newFor("Baz").build();
    parser.addArgument("--bar-text").required(true);
    parser.addArgument("--qux-text").required(true);
    parser.addArgument("--foo-text").required(true);
    parser.addArgument("--baz-text").required(true);

    Namespace res = parser.parseArgs(args);
    main(res.getString("bar_text"), res.getString("qux_text"), res.getString("foo_text"), res.getString("baz_text"));
  }
}